### PR TITLE
improve music player design

### PR DESCRIPTION
**Pull request by theScrabi**
issue: https://github.com/owncloud/android/issues/3121

### CODE REVIEW

File: owncloudApp/src/main/res/layout/preview_audio_fragment.xml [hunk 1]
```diff
@@ -14,47 +14,55 @@
...
-        android:layout_width="match_parent"
-        android:layout_height="0dp"
         android:layout_alignParentTop="true"
+        android:layout_centerHorizontal="true"
         android:layout_above="@+id/media_controller"
-        >
+        android:layout_gravity="center"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:padding="40dp">
```
Comment (abelgardep) [inline owncloudApp/src/main/res/layout/preview_audio_fragment.xml:39]: I'm not a huge fan of this. Adding a constraint layout just to include an image view may not be the best option. If you need some constraints, please move the whole layout to constraint layout and get rid of the relative one 👍

File: owncloudApp/src/main/res/layout-land/preview_audio_fragment.xml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:id="@+id/top"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:gravity="center"
+    tools:context=".ui.preview.PreviewAudioFragment"
+    android:filterTouchesWhenObscured="true">
+
+
+    <androidx.constraintlayout.widget.ConstraintLayout
```
Comment (abelgardep) [inline owncloudApp/src/main/res/layout-land/preview_audio_fragment.xml:31]: I think that we could get the same behaviour using the proper constraints in the portrait layout. That way, we could get rid of this second layout that we will need to maintain. What do you think?

File: owncloudApp/src/main/res/drawable/ic_music_note_eighth.xml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="200dp"
+    android:height="200dp"
+    android:viewportWidth="200"
+    android:viewportHeight="200">
+  <path
+      android:pathData="M60,8V113.5C54.1,110.1 47.3,108 40,108 17.9,108 0,125.9 0,148c0,22.1 17.9,40 40,40 22.1,0 40,-17.9 40,-40V48c33.102,0 66.203,0 99.305,0v67.332c-5.9,-3.4 -12.7,-5.5 -20,-5.5 -22.1,0 -40,17.9 -40,40 0,22.1 17.9,40 40,40 22.1,0 40,-17.9 40,-40V8C152.87,8 106.435,8 60,8Z"
+      android:strokeWidth="10"
+      android:fillColor="#cccccc"/>
```
Comment (abelgardep) [inline owncloudApp/src/main/res/drawable/ic_music_note_eighth.xml:9]: How did you choose that color? What do you think about using the app one? Could it be interesting to make it brandable?

**Review by abelgardep — CHANGES_REQUESTED** [review]
Some questions here @theScrabi.

BTW, do not forget to add the changelog item as described [here](https://github.com/owncloud/android/tree/master/changelog)

File: owncloudApp/src/main/res/drawable/ic_music_note_eighth.xml (hunk 3 above)
Comment (theScrabi) [reply owncloudApp/src/main/res/drawable/ic_music_note_eighth.xml:9]: Using our App colorpallet would make sense, however I don't think branding the placeholder for the music cover is very useful.

File: owncloudApp/src/main/res/layout/preview_audio_fragment.xml (hunk 1 above)
Comment (theScrabi) [reply owncloudApp/src/main/res/layout/preview_audio_fragment.xml]: done.

File: owncloudApp/src/main/res/layout-land/preview_audio_fragment.xml (hunk 2 above)
Comment (theScrabi) [reply owncloudApp/src/main/res/layout-land/preview_audio_fragment.xml]: I don't think so. We need a different parameter for making the image view rectangular.

File: owncloudApp/src/main/res/layout/preview_audio_fragment.xml [hunk 4]
```diff
@@ -29,32 +29,51 @@
 
     <FrameLayout
         android:id="@+id/visual_area"
-        android:layout_width="match_parent"
+        android:layout_width="0dp"
         android:layout_height="0dp"
-        android:layout_alignParentTop="true"
         android:layout_above="@+id/media_controller"
-        >
+        android:layout_alignParentTop="true"
```
Comment (abelgardep) [inline owncloudApp/src/main/res/layout/preview_audio_fragment.xml:35]: `layout_above` and `layout_alignParentTop` are not needed anymore, are they? We moved to ConstraintLayout 🍻

File: owncloudApp/src/main/res/layout/preview_audio_fragment.xml [hunk 5]
```diff
@@ -29,32 +29,51 @@
...
             android:id="@+id/image_preview"
             android:layout_width="match_parent"
             android:layout_height="match_parent"
-            android:layout_margin="@dimen/standard_margin"
-            android:layout_gravity="center"
             android:contentDescription="@string/preview_image_description"
-            android:src="@drawable/logo" />
+            android:scaleType="centerCrop"
+            android:src="@drawable/ic_place_holder_music_cover_art"
+            app:layout_constraintBottom_toBottomOf="parent" />
```
Comment (abelgardep) [inline owncloudApp/src/main/res/layout/preview_audio_fragment.xml:54]: The parent is not a Constraint layout, we don't need this constraint here, am I right?

File: owncloudApp/src/main/res/layout/preview_audio_fragment.xml [hunk 6]
```diff
@@ -29,32 +29,51 @@
...
+    <!-- last to draw, will be top in Z axis -->
     <com.owncloud.android.media.MediaControlView
         android:id="@id/media_controller"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
         android:layout_alignParentBottom="true"
         android:layout_margin="16dp"
-        />
+        app:layout_constraintBottom_toBottomOf="parent"
+        tools:layout_editor_absoluteX="16dp" />
```
Comment (abelgardep) [inline owncloudApp/src/main/res/layout/preview_audio_fragment.xml:75]: Do we need that `tools:layout_editor_absoluteX`?

File: owncloudApp/src/main/res/layout/preview_audio_fragment.xml [hunk 7]
```diff
@@ -29,32 +29,51 @@
...
-        android:layout_width="match_parent"
+        android:layout_width="0dp"
         android:layout_height="0dp"
-        android:layout_alignParentTop="true"
         android:layout_above="@+id/media_controller"
-        >
+        android:layout_alignParentTop="true"
+        android:layout_centerHorizontal="true"
+        android:padding="40dp"
+        app:layout_constraintDimensionRatio="H,1:1"
```
Comment (abelgardep) [inline owncloudApp/src/main/res/layout/preview_audio_fragment.xml:38]: If we reduce the padding to 16dp (Same as media_controller) and we move that `layout_constraintDimensionRatio="H,1:1"` to `layout_constraintDimensionRatio="1:1"` we would have the same behavior in landscape and we could remove the specific `preview_audio_fragment` for landscape. What do you think?

**Review by abelgardep — CHANGES_REQUESTED** [review]
Some questions more @theScrabi

File: owncloudApp/src/main/res/layout/preview_audio_fragment.xml (hunk 7 above)
Comment (theScrabi) [reply owncloudApp/src/main/res/layout/preview_audio_fragment.xml:38]: I didn't know I could leave out the `H` parameter.
However I don't like `16dp` on portrait, as the cover looks to huge:
|16dp|40dp|
|---|---|
|bla|asdf|

Comment (theScrabi) [reply owncloudApp/src/main/res/layout/preview_audio_fragment.xml]: All right I've settled with 30dp for land/portrait. :)

**Review by abelgardep — APPROVED** [review]
Approved on my side, Ready to QA @jesmrec

**Comment by jesmrec** [comment]
Checked with Android 7 & Android 11, new background image correctly displayed in both and both orientations.

Not much to test here

Approved

**Pull request merged**